src/Psalm/Internal/Codebase/ClassLikes.php

```diff
@@ -675,7 +676,23 @@ public function classImplements(string $fq_class_name, string $interface): bool
         }
         $class_storage = $this->classlike_storage_provider->get($fq_class_name);
 
-        return isset($class_storage->class_implements[$interface_id]);
+        if (isset($class_storage->class_implements[$interface_id])) {
+            return true;
+        }
+
+        foreach (array_keys($class_storage->class_implements) as $implementing_interface_lc) {
```

Comment:
```suggestion
        foreach ($class_storage->class_implements as $implementing_interface_lc => $_) {
```
is marginally faster and is just as clear, IMHO